addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3808,8 +3808,9 @@ export class OdooEditor extends EventTarget {
             }
             // Find previous character.
             let previousCharacter = focusOffset > 0 && focusNode.textContent[focusOffset - 1];
-            if (!previousCharacter) {
-                focusNode = previousLeaf(focusNode, this.editable);
+            const previousNode = previousLeaf(focusNode, this.editable);
+            if (!previousCharacter && closestBlock(previousNode) === closestBlock(focusNode)) {
```

Comment:
We don't know if `previousNode` is defined so I'd like this check to explicitly check it before calling `closestBlock` on it. One example of pernicious issue with this implicit check is that if both `previousNode` and `focusNode` are undefined, `closestBlock(previousNode) === closestBlock(focusNode)` will return true.